modin/pandas/test/test_groupby.py

```diff
@@ -1840,51 +1850,109 @@ def test_handle_as_index(
             "https://github.com/pandas-dev/pandas/issues/36698"
         )
 
-    data = {
-        "a": [1, 1, 2, 2],
-        "b": [2, 2, 3, 3],
-        "c": [1, 2, 3, 4],
-        "d": [5, 6, 7, 8],
-    }
+    df = pandas.DataFrame(test_groupby_data)
+    external_by_cols = GroupBy.validate_by(df.add_prefix("external_"))
+
+    if has_categorical_by:
+        df = df.astype({df.columns[0]: "category"})
 
     if isinstance(agg_func, dict):
+        agg_func = {df.columns[key]: value for key, value in agg_func.items()}
         selection = list(agg_func.keys())
         agg_dict = agg_func
         agg_func = lambda grp: grp.agg(agg_dict)  # noqa: E731 (lambda assignment)
     else:
         selection = None
 
-    df = pandas.DataFrame(data)
-    external_by_cols = GroupBy.validate_by(df.add_prefix("external_"))
+    # Selecting 'by' columns from both sides of the frame so they located in different partitions
+    internal_by = df.columns[
+        [i if i % 2 == 0 else -i for i in range(internal_by_length)]
+    ].tolist()
```

Comment:
are you telling me this clever way is better than
```suggestion
    internal_by = df.columns[range(-internal_by_length // 2, internal_by_length // 2)].tolist()
```
? sure, my suggestion takes consecutive columns from the beginning and end and in different order, but does it matter?